src/figuresClasses.ts

```diff
@@ -1,19 +1,95 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.isValidTriangle()) {
+      throw new Error(`Sides ${a}, ${b} and ${c} can't form a triangle`);
+    }
+  }
 
-export class Circle {
+  isValidTriangle: () => boolean = () => {
```

Comment:
Good method, let's make it private, cause we are using it only internally

Reply:
If the method has name `isValidTriangle`, it should return `true` if it's value, and `false` if it's invalid